service/authorization/v2/authorization.proto

```diff
@@ -0,0 +1,224 @@
+message EntityEntitlements {
+  message ActionsList {
+    repeated policy.Action actions = 1;
+  }
+
+  // ephemeral id for tracking between request and response
+  string ephemeral_id = 1;
+  map<string, ActionsList> actions_per_attribute_value_fqn = 2;
+}
+
+// Either a set of attribute values (such as those on a TDF) or a registered resource value
+message Resource {
+  // ephemeral id for tracking between request and response
+  string ephemeral_id = 1;
+
+  message AttributeValues {
+    repeated string fqns = 1;
+  }
+
+  oneof resource {
+    AttributeValues attribute_values = 2;
+    string registered_resource_value_fqn = 3;
+  }
+}
+
+message ResourceDecision {
+  // ephemeral id for tracking between request and response
+  string ephemeral_resource_id = 1;
+  // the decision result
+  Decision decision = 3;
+}
+
+// Can the entity (PE or NPE) access?
+// 1. one entity chain
+// 2. one action
+// 3. one resource
+message GetDecisionRequest {
+  authorization.EntityChain entity = 1 [(buf.validate.field).required = true];
+  // name on action is required
+  policy.Action action = 2 [(buf.validate.field).required = true];
+  Resource resource = 3 [(buf.validate.field).required = true];
+
+  option (buf.validate.message).cel = {
+    id: "get_decision_request.action_name_required"
+    message: "action.name must be provided"
+    expression: "has(this.action.name)"
+  };
+}
+message GetDecisionResponse {
+  // ephemeral id for tracking between request and response
+  string ephemeral_entity_chain_id = 1;
+  // name on action is required
+  policy.Action action = 2;
+  // the decision on the resource
+  ResourceDecision decision = 3;
+  // optional list of obligations represented in URI format
+  // repeated string obligations = 4;
+}
+
+// Can the entity (PE or NPE) access?
+// 1. one entity chain
+// 2. one action
+// 3. multiple resources
+// Note: this is a more performant bulk request for multiple resource decisions
+message GetDecisionMultiResourceRequest {
+  // The entity chain must be defined
+  authorization.EntityChain entity = 1 [(buf.validate.field).required = true];
+  // name on action is required
+  policy.Action action = 2 [(buf.validate.field).required = true];
+  repeated Resource resources = 3 [
+    (buf.validate.field).required = true,
+    (buf.validate.field).repeated = {min_items: 1}
+  ];
+
+  option (buf.validate.message).cel = {
+    id: "get_decision_multi_request.action_name_required"
+    message: "action.name must be provided"
+    expression: "has(this.action.name)"
+  };
+}
+message GetDecisionMultiResourceResponse {
+  // ephemeral id for tracking between request and response
+  string ephemeral_entity_chain_id = 1;
+  // name on action is to be relied upon
+  policy.Action action = 2;
+  // convenience flag indicating global resource decisions result (permit/deny)
+  google.protobuf.BoolValue all_permitted = 3;
+  // individual resource decisions
+  repeated ResourceDecision resource_decisions = 4;
```

Comment:
i guess this is where you need the ephemeral resource ids in the decision responses, when you have many of them. but do you need the ephemeral_entity_chain_id?

Reply:
clean up the response to omit what is just repeating the request fields right back